// @mem-labs/mem-node/dist/utils/logger/Logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Logger = void 0;
const constants_1 = require("./constants");
const types_1 = require("./types");
/**
 * Default logger which logs to stdout and stderr.
 */
class Logger {
    constructor({ level }) {
        this.level = level !== null && level !== void 0 ? level : types_1.LogLevel.INFO;
    }
    /**
     * Log a debug message.
     */
    debug(...msg) {
        if (this.shouldLogSeverity(types_1.LogLevel.DEBUG)) {
            console.debug(`[debug]`, ...msg);
        }
    }
    /**
     * Log an info message.
     */
    info(...msg) {
        if (this.shouldLogSeverity(types_1.LogLevel.INFO)) {
            console.info(`[info]`, ...msg);
        }
    }
    /**
     * Log a warning message.
     */
    warn(...msg) {
        if (this.shouldLogSeverity(types_1.LogLevel.WARN)) {
            console.warn(`[warn]`, ...msg);
        }
    }
    /**
     * Log an error message
     */
    error(...msg) {
        if (this.shouldLogSeverity(types_1.LogLevel.ERROR)) {
            console.error(`[error]`, ...msg);
        }
    }
    /**
     * Check if the inputLevel is >= the current log level.
     */
    shouldLogSeverity(inputLevel) {
        const minimumSeverity = constants_1.logLevelSeverities[this.level];
        const inputSeverity = constants_1.logLevelSeverities[inputLevel];
        return inputSeverity >= minimumSeverity;
    }
}
exports.Logger = Logger;
