// @mem-labs/mem-node/dist/api/__codegen__/types.js
"use strict";
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/ban-types */
/**
 * This file is automatically generated by graphql-code-generator.
 * Do not edit it manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HealthCheckDocument = exports.CreateMemDocument = exports.BatchCreateMemsDocument = exports.GqTrackedEventSlug = exports.GqRedisKey = exports.GqPlatformPlanKind = exports.GqPlatformFeatureKind = exports.GqMemFormat = exports.GqMemCreationSourceKind = exports.GqMemCreationImportSourceKind = exports.GqKnowledgeNodeSearchObjectKind = exports.GqKnowledgeGraphRecommendedActionType = exports.GqKnowledgeGraphFacetKind = exports.GqKnowledgeGraphAttributesForFaceting = exports.GqGoogleOAuthClient = exports.GqFeatureToggleSlug = exports.GqFavoriteItemKind = exports.GqFactType = exports.GqConvertedObjectIDsAfterSearchEventNames = exports.GqCloudTaskKind = exports.GqClientEventKind = exports.GqClickedEntityEventNames = exports.GqAdminFunctionName = exports.GqAccountWorkspaceRoleKind = exports.GqAccountSearchKnowledgeGraphSources = exports.GqAccountPreferenceSlug = exports.GqAccountMemInboxSubscriptionLevelKind = exports.GqAccountMemInboxStateKind = exports.GqAccountFavoriteItemKind = void 0;
/** The kind of account favorite item. */
var GqAccountFavoriteItemKind;
(function (GqAccountFavoriteItemKind) {
    GqAccountFavoriteItemKind["Mem"] = "MEM";
    GqAccountFavoriteItemKind["SearchQuery"] = "SEARCH_QUERY";
    GqAccountFavoriteItemKind["Topic"] = "TOPIC";
})(GqAccountFavoriteItemKind = exports.GqAccountFavoriteItemKind || (exports.GqAccountFavoriteItemKind = {}));
/** A list of valid inbox account state kinds. */
var GqAccountMemInboxStateKind;
(function (GqAccountMemInboxStateKind) {
    GqAccountMemInboxStateKind["Inbox"] = "INBOX";
    GqAccountMemInboxStateKind["Archived"] = "ARCHIVED";
    GqAccountMemInboxStateKind["Snoozed"] = "SNOOZED";
})(GqAccountMemInboxStateKind = exports.GqAccountMemInboxStateKind || (exports.GqAccountMemInboxStateKind = {}));
/** A list of valid inbox subscription level kinds. */
var GqAccountMemInboxSubscriptionLevelKind;
(function (GqAccountMemInboxSubscriptionLevelKind) {
    GqAccountMemInboxSubscriptionLevelKind["All"] = "ALL";
    GqAccountMemInboxSubscriptionLevelKind["Digest"] = "DIGEST";
    GqAccountMemInboxSubscriptionLevelKind["Muted"] = "MUTED";
})(GqAccountMemInboxSubscriptionLevelKind = exports.GqAccountMemInboxSubscriptionLevelKind || (exports.GqAccountMemInboxSubscriptionLevelKind = {}));
/** A list of valid account preference slugs. */
var GqAccountPreferenceSlug;
(function (GqAccountPreferenceSlug) {
    GqAccountPreferenceSlug["ExampleAccountPreference"] = "EXAMPLE_ACCOUNT_PREFERENCE";
    GqAccountPreferenceSlug["RecordMemInteractionLocationPolicy"] = "RECORD_MEM_INTERACTION_LOCATION_POLICY";
})(GqAccountPreferenceSlug = exports.GqAccountPreferenceSlug || (exports.GqAccountPreferenceSlug = {}));
var GqAccountSearchKnowledgeGraphSources;
(function (GqAccountSearchKnowledgeGraphSources) {
    GqAccountSearchKnowledgeGraphSources["MemApp"] = "MEM_APP";
    GqAccountSearchKnowledgeGraphSources["MemIos"] = "MEM_IOS";
    GqAccountSearchKnowledgeGraphSources["MemKeyboard"] = "MEM_KEYBOARD";
    GqAccountSearchKnowledgeGraphSources["MemShareSheet"] = "MEM_SHARE_SHEET";
})(GqAccountSearchKnowledgeGraphSources = exports.GqAccountSearchKnowledgeGraphSources || (exports.GqAccountSearchKnowledgeGraphSources = {}));
/** A list of valid account workspace roles. */
var GqAccountWorkspaceRoleKind;
(function (GqAccountWorkspaceRoleKind) {
    GqAccountWorkspaceRoleKind["Owner"] = "OWNER";
    GqAccountWorkspaceRoleKind["Admin"] = "ADMIN";
    GqAccountWorkspaceRoleKind["Member"] = "MEMBER";
})(GqAccountWorkspaceRoleKind = exports.GqAccountWorkspaceRoleKind || (exports.GqAccountWorkspaceRoleKind = {}));
/** A list of admin functions. */
var GqAdminFunctionName;
(function (GqAdminFunctionName) {
    GqAdminFunctionName["IndexAllMemsFromFirebase"] = "INDEX_ALL_MEMS_FROM_FIREBASE";
    GqAdminFunctionName["ReindexAllMemsInApiDb"] = "REINDEX_ALL_MEMS_IN_API_DB";
    GqAdminFunctionName["IndexAllTemplatesFromFirebase"] = "INDEX_ALL_TEMPLATES_FROM_FIREBASE";
    GqAdminFunctionName["BackfillAllMemEmbeddings"] = "BACKFILL_ALL_MEM_EMBEDDINGS";
    GqAdminFunctionName["BackfillMemCurieEmbeddingsForAuthorizedUsers"] = "BACKFILL_MEM_CURIE_EMBEDDINGS_FOR_AUTHORIZED_USERS";
    GqAdminFunctionName["BackfillMemCurieEmbeddingsForAuthorizedUsersFromDate"] = "BACKFILL_MEM_CURIE_EMBEDDINGS_FOR_AUTHORIZED_USERS_FROM_DATE";
    GqAdminFunctionName["BackfillMemCurieSearchDocEmbeddingsForAuthorizedUsers"] = "BACKFILL_MEM_CURIE_SEARCH_DOC_EMBEDDINGS_FOR_AUTHORIZED_USERS";
    GqAdminFunctionName["IndexAllFirestoreUsersInFirebase"] = "INDEX_ALL_FIRESTORE_USERS_IN_FIREBASE";
})(GqAdminFunctionName = exports.GqAdminFunctionName || (exports.GqAdminFunctionName = {}));
var GqClickedEntityEventNames;
(function (GqClickedEntityEventNames) {
    GqClickedEntityEventNames["MemClicked"] = "MEM_CLICKED";
})(GqClickedEntityEventNames = exports.GqClickedEntityEventNames || (exports.GqClickedEntityEventNames = {}));
/** A list of valid client events. */
var GqClientEventKind;
(function (GqClientEventKind) {
    GqClientEventKind["ExampleEvent"] = "EXAMPLE_EVENT";
    /** Generic */
    GqClientEventKind["MemIosAppBooted"] = "MEM_IOS_APP_BOOTED";
    GqClientEventKind["MemIosLoadPersistedAccountSuccess"] = "MEM_IOS_LOAD_PERSISTED_ACCOUNT_SUCCESS";
    GqClientEventKind["MemIosLoadPersistedAccountFailureNotFound"] = "MEM_IOS_LOAD_PERSISTED_ACCOUNT_FAILURE_NOT_FOUND";
    GqClientEventKind["MemIosAppOpened"] = "MEM_IOS_APP_OPENED";
    /** Login */
    GqClientEventKind["MemIosLoginScreenViewed"] = "MEM_IOS_LOGIN_SCREEN_VIEWED";
    GqClientEventKind["MemIosLoginScreenAccountLoginSuccess"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_LOGIN_SUCCESS";
    GqClientEventKind["MemIosLoginScreenAccountLoginFailureNotFound"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_LOGIN_FAILURE_NOT_FOUND";
    GqClientEventKind["MemIosLoginScreenAccountLoginFailureUnknown"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_LOGIN_FAILURE_UNKNOWN";
    GqClientEventKind["MemIosLoginScreenSignInWithGoogleClicked"] = "MEM_IOS_LOGIN_SCREEN_SIGN_IN_WITH_GOOGLE_CLICKED";
    GqClientEventKind["MemIosLoginScreenLearnMoreClicked"] = "MEM_IOS_LOGIN_SCREEN_LEARN_MORE_CLICKED";
    GqClientEventKind["MemIosLoginScreenAccountNotFoundMustSignUpClicked"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_NOT_FOUND_MUST_SIGN_UP_CLICKED";
    /** Home */
    GqClientEventKind["MemIosHomeScreenViewed"] = "MEM_IOS_HOME_SCREEN_VIEWED";
    GqClientEventKind["MemIosHomeScreenLoadedResults"] = "MEM_IOS_HOME_SCREEN_LOADED_RESULTS";
    GqClientEventKind["MemIosHomeScreenSawNoResults"] = "MEM_IOS_HOME_SCREEN_SAW_NO_RESULTS";
    GqClientEventKind["MemIosHomeScreenLoadedMoreResults"] = "MEM_IOS_HOME_SCREEN_LOADED_MORE_RESULTS";
    GqClientEventKind["MemIosHomeScreenRefreshedResults"] = "MEM_IOS_HOME_SCREEN_REFRESHED_RESULTS";
    GqClientEventKind["MemIosHomeScreenSuggestionClicked"] = "MEM_IOS_HOME_SCREEN_SUGGESTION_CLICKED";
    /** Search */
    GqClientEventKind["MemIosSearchScreenViewed"] = "MEM_IOS_SEARCH_SCREEN_VIEWED";
    GqClientEventKind["MemIosSearchScreenSearchExecuted"] = "MEM_IOS_SEARCH_SCREEN_SEARCH_EXECUTED";
    GqClientEventKind["MemIosSearchScreenSawNoResults"] = "MEM_IOS_SEARCH_SCREEN_SAW_NO_RESULTS";
    GqClientEventKind["MemIosSearchScreenLoadedMoreResults"] = "MEM_IOS_SEARCH_SCREEN_LOADED_MORE_RESULTS";
    GqClientEventKind["MemIosSearchScreenSuggestionClicked"] = "MEM_IOS_SEARCH_SCREEN_SUGGESTION_CLICKED";
    GqClientEventKind["MemIosSearchScreenOpenedFromNavigationBar"] = "MEM_IOS_SEARCH_SCREEN_OPENED_FROM_NAVIGATION_BAR";
    GqClientEventKind["MemIosSearchScreenOpenedFromHomeBar"] = "MEM_IOS_SEARCH_SCREEN_OPENED_FROM_HOME_BAR";
    /** Share Sheet */
    GqClientEventKind["MemIosShareSheetViewed"] = "MEM_IOS_SHARE_SHEET_VIEWED";
    /** Slice Results (Home + Search Views) */
    GqClientEventKind["MemIosSliceResultsItemClicked"] = "MEM_IOS_SLICE_RESULTS_ITEM_CLICKED";
    /** Details */
    GqClientEventKind["MemIosDetailScreenViewed"] = "MEM_IOS_DETAIL_SCREEN_VIEWED";
    GqClientEventKind["MemIosDetailScreenMemEdited"] = "MEM_IOS_DETAIL_SCREEN_MEM_EDITED";
    GqClientEventKind["MemIosDetailScreenMemComposed"] = "MEM_IOS_DETAIL_SCREEN_MEM_COMPOSED";
    GqClientEventKind["MemIosDetailScreenTopicClicked"] = "MEM_IOS_DETAIL_SCREEN_TOPIC_CLICKED";
    GqClientEventKind["MemIosDetailScreenWebLinkClicked"] = "MEM_IOS_DETAIL_SCREEN_WEB_LINK_CLICKED";
    GqClientEventKind["MemIosDetailScreenLinkedFirestoreNoteClicked"] = "MEM_IOS_DETAIL_SCREEN_LINKED_FIRESTORE_NOTE_CLICKED";
    GqClientEventKind["MemIosDetailScreenMentionedFirestoreUserClicked"] = "MEM_IOS_DETAIL_SCREEN_MENTIONED_FIRESTORE_USER_CLICKED";
    GqClientEventKind["MemIosDetailScreenMentionedFirestoreGroupClicked"] = "MEM_IOS_DETAIL_SCREEN_MENTIONED_FIRESTORE_GROUP_CLICKED";
    GqClientEventKind["MemIosDetailScreenRelatedMemsViewed"] = "MEM_IOS_DETAIL_SCREEN_RELATED_MEMS_VIEWED";
    /** Editing */
    GqClientEventKind["MemIosEditorCarouselCameraClicked"] = "MEM_IOS_EDITOR_CAROUSEL_CAMERA_CLICKED";
    GqClientEventKind["MemIosEditorCarouselPhotoClicked"] = "MEM_IOS_EDITOR_CAROUSEL_PHOTO_CLICKED";
    GqClientEventKind["MemIosEditorCarouselGalleryClicked"] = "MEM_IOS_EDITOR_CAROUSEL_GALLERY_CLICKED";
    GqClientEventKind["MemIosEditorToolbarGalleryClicked"] = "MEM_IOS_EDITOR_TOOLBAR_GALLERY_CLICKED";
    GqClientEventKind["MemIosEditorToolbarBulletedListClicked"] = "MEM_IOS_EDITOR_TOOLBAR_BULLETED_LIST_CLICKED";
    GqClientEventKind["MemIosEditorToolbarChecklistClicked"] = "MEM_IOS_EDITOR_TOOLBAR_CHECKLIST_CLICKED";
    GqClientEventKind["MemIosEditorToolbarAddTagClicked"] = "MEM_IOS_EDITOR_TOOLBAR_ADD_TAG_CLICKED";
    GqClientEventKind["MemIosEditorToolbarFormatClicked"] = "MEM_IOS_EDITOR_TOOLBAR_FORMAT_CLICKED";
    GqClientEventKind["MemIosEditorToolbarH1Clicked"] = "MEM_IOS_EDITOR_TOOLBAR_H1_CLICKED";
    GqClientEventKind["MemIosEditorToolbarH2Clicked"] = "MEM_IOS_EDITOR_TOOLBAR_H2_CLICKED";
    GqClientEventKind["MemIosEditorToolbarBoldClicked"] = "MEM_IOS_EDITOR_TOOLBAR_BOLD_CLICKED";
    GqClientEventKind["MemIosEditorToolbarItalicClicked"] = "MEM_IOS_EDITOR_TOOLBAR_ITALIC_CLICKED";
    GqClientEventKind["MemIosEditorToolbarUnderlineClicked"] = "MEM_IOS_EDITOR_TOOLBAR_UNDERLINE_CLICKED";
    GqClientEventKind["MemIosEditorToolbarStrikethroughClicked"] = "MEM_IOS_EDITOR_TOOLBAR_STRIKETHROUGH_CLICKED";
    GqClientEventKind["MemIosEditorToolbarCodeBlockClicked"] = "MEM_IOS_EDITOR_TOOLBAR_CODE_BLOCK_CLICKED";
    GqClientEventKind["MemIosEditorToolbarQuoteBlockClicked"] = "MEM_IOS_EDITOR_TOOLBAR_QUOTE_BLOCK_CLICKED";
    GqClientEventKind["MemIosEditorToolbarRemoveFormattingClicked"] = "MEM_IOS_EDITOR_TOOLBAR_REMOVE_FORMATTING_CLICKED";
    /** Side Navigation */
    GqClientEventKind["MemIosSideNavigationViewed"] = "MEM_IOS_SIDE_NAVIGATION_VIEWED";
    /** Help And Support */
    GqClientEventKind["MemIosSupportScreenViewed"] = "MEM_IOS_SUPPORT_SCREEN_VIEWED";
    GqClientEventKind["MemIosSupportScreenProductUpdatesClicked"] = "MEM_IOS_SUPPORT_SCREEN_PRODUCT_UPDATES_CLICKED";
    GqClientEventKind["MemIosSupportScreenFaqsClicked"] = "MEM_IOS_SUPPORT_SCREEN_FAQS_CLICKED";
    GqClientEventKind["MemIosSupportScreenTutorialsClicked"] = "MEM_IOS_SUPPORT_SCREEN_TUTORIALS_CLICKED";
    GqClientEventKind["MemIosSupportScreenBlogClicked"] = "MEM_IOS_SUPPORT_SCREEN_BLOG_CLICKED";
    GqClientEventKind["MemIosSupportScreenContactUsClicked"] = "MEM_IOS_SUPPORT_SCREEN_CONTACT_US_CLICKED";
    /** Other Events */
    GqClientEventKind["MemIosAccountLoggedOut"] = "MEM_IOS_ACCOUNT_LOGGED_OUT";
})(GqClientEventKind = exports.GqClientEventKind || (exports.GqClientEventKind = {}));
/** A list of valid cloud tasks. */
var GqCloudTaskKind;
(function (GqCloudTaskKind) {
    GqCloudTaskKind["HealthCheck"] = "HEALTH_CHECK";
    GqCloudTaskKind["ExampleEvent"] = "EXAMPLE_EVENT";
    GqCloudTaskKind["IndexMemEntity"] = "INDEX_MEM_ENTITY";
    GqCloudTaskKind["ReprocessMemEntity"] = "REPROCESS_MEM_ENTITY";
    GqCloudTaskKind["ExtractPeopleFromMemEntity"] = "EXTRACT_PEOPLE_FROM_MEM_ENTITY";
    GqCloudTaskKind["ExtractImplicitTopicsFromMemEntity"] = "EXTRACT_IMPLICIT_TOPICS_FROM_MEM_ENTITY";
    GqCloudTaskKind["ExtractTextFromImagesInsideMemEntity"] = "EXTRACT_TEXT_FROM_IMAGES_INSIDE_MEM_ENTITY";
    GqCloudTaskKind["StoreTrackedEvent"] = "STORE_TRACKED_EVENT";
    GqCloudTaskKind["ReindexAllMemsForAccount"] = "REINDEX_ALL_MEMS_FOR_ACCOUNT";
    GqCloudTaskKind["ReindexAllMemsInApiDb"] = "REINDEX_ALL_MEMS_IN_API_DB";
    GqCloudTaskKind["IndexAllMemsInFirebase"] = "INDEX_ALL_MEMS_IN_FIREBASE";
    GqCloudTaskKind["CreateUserAppNotificationForAllUsers"] = "CREATE_USER_APP_NOTIFICATION_FOR_ALL_USERS";
    GqCloudTaskKind["BatchIndexMemEntities"] = "BATCH_INDEX_MEM_ENTITIES";
    GqCloudTaskKind["IndexAccountEntity"] = "INDEX_ACCOUNT_ENTITY";
    GqCloudTaskKind["IndexAllUsersInFirebase"] = "INDEX_ALL_USERS_IN_FIREBASE";
    GqCloudTaskKind["BatchIndexFirestoreUsers"] = "BATCH_INDEX_FIRESTORE_USERS";
    GqCloudTaskKind["IndexGroupEntity"] = "INDEX_GROUP_ENTITY";
    GqCloudTaskKind["IndexMemTemplateEntity"] = "INDEX_MEM_TEMPLATE_ENTITY";
    GqCloudTaskKind["IndexAllTemplatesInFirebase"] = "INDEX_ALL_TEMPLATES_IN_FIREBASE";
    GqCloudTaskKind["BatchImportMems"] = "BATCH_IMPORT_MEMS";
    GqCloudTaskKind["ImportMemTask"] = "IMPORT_MEM_TASK";
    GqCloudTaskKind["BatchIndexMemTemplateEntities"] = "BATCH_INDEX_MEM_TEMPLATE_ENTITIES";
    GqCloudTaskKind["BackfillAllMemEmbeddings"] = "BACKFILL_ALL_MEM_EMBEDDINGS";
    GqCloudTaskKind["BackfillMemCurieEmbeddingsForAuthorizedUsers"] = "BACKFILL_MEM_CURIE_EMBEDDINGS_FOR_AUTHORIZED_USERS";
    GqCloudTaskKind["BackfillMemCurieSearchDocEmbeddingsForAuthorizedUsers"] = "BACKFILL_MEM_CURIE_SEARCH_DOC_EMBEDDINGS_FOR_AUTHORIZED_USERS";
    GqCloudTaskKind["BackfillMemCurieEmbeddingsForAuthorizedUsersFromDate"] = "BACKFILL_MEM_CURIE_EMBEDDINGS_FOR_AUTHORIZED_USERS_FROM_DATE";
    GqCloudTaskKind["BatchEmbedMems"] = "BATCH_EMBED_MEMS";
    GqCloudTaskKind["EmbedMem"] = "EMBED_MEM";
    GqCloudTaskKind["EmbedMemSearchDoc"] = "EMBED_MEM_SEARCH_DOC";
    GqCloudTaskKind["EmbedMems"] = "EMBED_MEMS";
    GqCloudTaskKind["EmbedMemsSearchDoc"] = "EMBED_MEMS_SEARCH_DOC";
    GqCloudTaskKind["BackfillMemCurieEmbeddingsForUserUsingFirestoreUserId"] = "BACKFILL_MEM_CURIE_EMBEDDINGS_FOR_USER_USING_FIRESTORE_USER_ID";
    GqCloudTaskKind["BackfillMemCurieSearchDocEmbeddingsForUserUsingFirestoreUserId"] = "BACKFILL_MEM_CURIE_SEARCH_DOC_EMBEDDINGS_FOR_USER_USING_FIRESTORE_USER_ID";
})(GqCloudTaskKind = exports.GqCloudTaskKind || (exports.GqCloudTaskKind = {}));
var GqConvertedObjectIDsAfterSearchEventNames;
(function (GqConvertedObjectIDsAfterSearchEventNames) {
    GqConvertedObjectIDsAfterSearchEventNames["MemConverted"] = "MEM_CONVERTED";
})(GqConvertedObjectIDsAfterSearchEventNames = exports.GqConvertedObjectIDsAfterSearchEventNames || (exports.GqConvertedObjectIDsAfterSearchEventNames = {}));
var GqFactType;
(function (GqFactType) {
    GqFactType["Numeric"] = "NUMERIC";
    GqFactType["Email"] = "EMAIL";
    GqFactType["Url"] = "URL";
    GqFactType["Password"] = "PASSWORD";
})(GqFactType = exports.GqFactType || (exports.GqFactType = {}));
/**
 * The kind of favorite item.
 * @deprecated - use AccountFavoriteItemKind instead.
 */
var GqFavoriteItemKind;
(function (GqFavoriteItemKind) {
    GqFavoriteItemKind["Mem"] = "MEM";
    GqFavoriteItemKind["SearchQuery"] = "SEARCH_QUERY";
    GqFavoriteItemKind["Topic"] = "TOPIC";
})(GqFavoriteItemKind = exports.GqFavoriteItemKind || (exports.GqFavoriteItemKind = {}));
/** A list of valid feature toggle slugs. */
var GqFeatureToggleSlug;
(function (GqFeatureToggleSlug) {
    GqFeatureToggleSlug["ExampleFeature"] = "EXAMPLE_FEATURE";
    GqFeatureToggleSlug["MemIosAlphaTester"] = "MEM_IOS_ALPHA_TESTER";
    GqFeatureToggleSlug["SearchIndexingAlphaTester"] = "SEARCH_INDEXING_ALPHA_TESTER";
    GqFeatureToggleSlug["MemIosBroadcastOnClient"] = "MEM_IOS_BROADCAST_ON_CLIENT";
    GqFeatureToggleSlug["SimilarMemsFeature"] = "SIMILAR_MEMS_FEATURE";
    GqFeatureToggleSlug["WorkspacesAlphaTester"] = "WORKSPACES_ALPHA_TESTER";
    GqFeatureToggleSlug["ElasticCloudSearchTester"] = "ELASTIC_CLOUD_SEARCH_TESTER";
})(GqFeatureToggleSlug = exports.GqFeatureToggleSlug || (exports.GqFeatureToggleSlug = {}));
var GqGoogleOAuthClient;
(function (GqGoogleOAuthClient) {
    GqGoogleOAuthClient["MemWebClient"] = "MEM_WEB_CLIENT";
    GqGoogleOAuthClient["MemAdminPortal"] = "MEM_ADMIN_PORTAL";
    GqGoogleOAuthClient["MemIos"] = "MEM_IOS";
})(GqGoogleOAuthClient = exports.GqGoogleOAuthClient || (exports.GqGoogleOAuthClient = {}));
/** Extra filterable facets @todo, find a better way to do this */
var GqKnowledgeGraphAttributesForFaceting;
(function (GqKnowledgeGraphAttributesForFaceting) {
    GqKnowledgeGraphAttributesForFaceting["PermittedFirestoreUserIds"] = "permittedFirestoreUserIds";
    GqKnowledgeGraphAttributesForFaceting["PermittedFirestoreUserIdsString"] = "permittedFirestoreUserIdsString";
})(GqKnowledgeGraphAttributesForFaceting = exports.GqKnowledgeGraphAttributesForFaceting || (exports.GqKnowledgeGraphAttributesForFaceting = {}));
/** Filterable facets for our knowledge graph. */
var GqKnowledgeGraphFacetKind;
(function (GqKnowledgeGraphFacetKind) {
    GqKnowledgeGraphFacetKind["KnowledgeNodeKind"] = "KNOWLEDGE_NODE_KIND";
    GqKnowledgeGraphFacetKind["TopicLabels"] = "TOPIC_LABELS";
    GqKnowledgeGraphFacetKind["PermittedGroups"] = "PERMITTED_GROUPS";
    GqKnowledgeGraphFacetKind["EditedByUsers"] = "EDITED_BY_USERS";
    GqKnowledgeGraphFacetKind["LinkKinds"] = "LINK_KINDS";
    GqKnowledgeGraphFacetKind["MediaKinds"] = "MEDIA_KINDS";
    GqKnowledgeGraphFacetKind["CapturedWithSources"] = "CAPTURED_WITH_SOURCES";
    GqKnowledgeGraphFacetKind["IsInInboxForAccounts"] = "IS_IN_INBOX_FOR_ACCOUNTS";
    GqKnowledgeGraphFacetKind["IsInTrashForAccounts"] = "IS_IN_TRASH_FOR_ACCOUNTS";
    GqKnowledgeGraphFacetKind["IsUnreadForAccounts"] = "IS_UNREAD_FOR_ACCOUNTS";
    GqKnowledgeGraphFacetKind["CreatedByAccount"] = "CREATED_BY_ACCOUNT";
    GqKnowledgeGraphFacetKind["PermittedFirestoreGroupsAndUsers"] = "PERMITTED_FIRESTORE_GROUPS_AND_USERS";
    GqKnowledgeGraphFacetKind["LastEditedByAccountId"] = "LAST_EDITED_BY_ACCOUNT_ID";
})(GqKnowledgeGraphFacetKind = exports.GqKnowledgeGraphFacetKind || (exports.GqKnowledgeGraphFacetKind = {}));
var GqKnowledgeGraphRecommendedActionType;
(function (GqKnowledgeGraphRecommendedActionType) {
    GqKnowledgeGraphRecommendedActionType["OpenMem"] = "OPEN_MEM";
})(GqKnowledgeGraphRecommendedActionType = exports.GqKnowledgeGraphRecommendedActionType || (exports.GqKnowledgeGraphRecommendedActionType = {}));
var GqKnowledgeNodeSearchObjectKind;
(function (GqKnowledgeNodeSearchObjectKind) {
    GqKnowledgeNodeSearchObjectKind["Mem"] = "MEM";
    GqKnowledgeNodeSearchObjectKind["Account"] = "ACCOUNT";
    GqKnowledgeNodeSearchObjectKind["Group"] = "GROUP";
    GqKnowledgeNodeSearchObjectKind["MemTemplate"] = "MEM_TEMPLATE";
    GqKnowledgeNodeSearchObjectKind["Fact"] = "FACT";
})(GqKnowledgeNodeSearchObjectKind = exports.GqKnowledgeNodeSearchObjectKind || (exports.GqKnowledgeNodeSearchObjectKind = {}));
var GqMemCreationImportSourceKind;
(function (GqMemCreationImportSourceKind) {
    GqMemCreationImportSourceKind["Notion"] = "Notion";
    GqMemCreationImportSourceKind["Obsidian"] = "Obsidian";
    GqMemCreationImportSourceKind["Markdown"] = "Markdown";
    GqMemCreationImportSourceKind["Roam"] = "Roam";
    GqMemCreationImportSourceKind["AppleNotes"] = "AppleNotes";
    GqMemCreationImportSourceKind["Bear"] = "Bear";
    GqMemCreationImportSourceKind["IaWriter"] = "IAWriter";
    GqMemCreationImportSourceKind["Evernote"] = "Evernote";
})(GqMemCreationImportSourceKind = exports.GqMemCreationImportSourceKind || (exports.GqMemCreationImportSourceKind = {}));
var GqMemCreationSourceKind;
(function (GqMemCreationSourceKind) {
    GqMemCreationSourceKind["App"] = "App";
    GqMemCreationSourceKind["Import"] = "Import";
    GqMemCreationSourceKind["Sms"] = "Sms";
    GqMemCreationSourceKind["Api"] = "API";
    GqMemCreationSourceKind["Unknown"] = "Unknown";
})(GqMemCreationSourceKind = exports.GqMemCreationSourceKind || (exports.GqMemCreationSourceKind = {}));
/** This is deprecated, and should no longer be used. */
var GqMemFormat;
(function (GqMemFormat) {
    GqMemFormat["Markdown"] = "MARKDOWN";
})(GqMemFormat = exports.GqMemFormat || (exports.GqMemFormat = {}));
/** Platform Features */
var GqPlatformFeatureKind;
(function (GqPlatformFeatureKind) {
    GqPlatformFeatureKind["Tags"] = "TAGS";
    GqPlatformFeatureKind["Templates"] = "TEMPLATES";
    GqPlatformFeatureKind["ConnectedEmails"] = "CONNECTED_EMAILS";
    GqPlatformFeatureKind["ConnectedCalendars"] = "CONNECTED_CALENDARS";
    GqPlatformFeatureKind["Texts"] = "TEXTS";
    GqPlatformFeatureKind["SmartSearch"] = "SMART_SEARCH";
    GqPlatformFeatureKind["SimilarMems"] = "SIMILAR_MEMS";
    GqPlatformFeatureKind["ApiKeys"] = "API_KEYS";
    GqPlatformFeatureKind["Zapier"] = "ZAPIER";
    GqPlatformFeatureKind["Groups"] = "GROUPS";
    GqPlatformFeatureKind["GroupMembers"] = "GROUP_MEMBERS";
    GqPlatformFeatureKind["FileStorage"] = "FILE_STORAGE";
    GqPlatformFeatureKind["FileUploads"] = "FILE_UPLOADS";
})(GqPlatformFeatureKind = exports.GqPlatformFeatureKind || (exports.GqPlatformFeatureKind = {}));
/** Platform Plans */
var GqPlatformPlanKind;
(function (GqPlatformPlanKind) {
    GqPlatformPlanKind["MemFree"] = "MEM_FREE";
    GqPlatformPlanKind["MemPro"] = "MEM_PRO";
    GqPlatformPlanKind["MemX"] = "MEM_X";
    GqPlatformPlanKind["MemTeams"] = "MEM_TEAMS";
})(GqPlatformPlanKind = exports.GqPlatformPlanKind || (exports.GqPlatformPlanKind = {}));
/** A list of common redis keys. */
var GqRedisKey;
(function (GqRedisKey) {
    GqRedisKey["NotificationGenerationProgress"] = "NOTIFICATION_GENERATION_PROGRESS";
    GqRedisKey["ReindexAllMemsInApiDbProgress"] = "REINDEX_ALL_MEMS_IN_API_DB_PROGRESS";
    GqRedisKey["Ping"] = "PING";
})(GqRedisKey = exports.GqRedisKey || (exports.GqRedisKey = {}));
/** A list of valid tracked event slugs. */
var GqTrackedEventSlug;
(function (GqTrackedEventSlug) {
    GqTrackedEventSlug["ExampleEvent"] = "EXAMPLE_EVENT";
    GqTrackedEventSlug["PersonIndexed"] = "PERSON_INDEXED";
    GqTrackedEventSlug["AccountIndexed"] = "ACCOUNT_INDEXED";
    GqTrackedEventSlug["GroupIndexed"] = "GROUP_INDEXED";
    GqTrackedEventSlug["MemTemplateIndexed"] = "MEM_TEMPLATE_INDEXED";
    GqTrackedEventSlug["MemCreated"] = "MEM_CREATED";
    GqTrackedEventSlug["MemContentAppended"] = "MEM_CONTENT_APPENDED";
    GqTrackedEventSlug["MemContentSynced"] = "MEM_CONTENT_SYNCED";
    GqTrackedEventSlug["MemContentTransformed"] = "MEM_CONTENT_TRANSFORMED";
    GqTrackedEventSlug["MemIndexed"] = "MEM_INDEXED";
    GqTrackedEventSlug["MemArchived"] = "MEM_ARCHIVED";
    /** Generic */
    GqTrackedEventSlug["MemIosAppBooted"] = "MEM_IOS_APP_BOOTED";
    GqTrackedEventSlug["MemIosLoadPersistedAccountSuccess"] = "MEM_IOS_LOAD_PERSISTED_ACCOUNT_SUCCESS";
    GqTrackedEventSlug["MemIosLoadPersistedAccountFailureNotFound"] = "MEM_IOS_LOAD_PERSISTED_ACCOUNT_FAILURE_NOT_FOUND";
    GqTrackedEventSlug["MemIosAppOpened"] = "MEM_IOS_APP_OPENED";
    /** Login */
    GqTrackedEventSlug["MemIosLoginScreenViewed"] = "MEM_IOS_LOGIN_SCREEN_VIEWED";
    GqTrackedEventSlug["MemIosLoginScreenAccountLoginSuccess"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_LOGIN_SUCCESS";
    GqTrackedEventSlug["MemIosLoginScreenAccountLoginFailureNotFound"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_LOGIN_FAILURE_NOT_FOUND";
    GqTrackedEventSlug["MemIosLoginScreenAccountLoginFailureUnknown"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_LOGIN_FAILURE_UNKNOWN";
    GqTrackedEventSlug["MemIosLoginScreenSignInWithGoogleClicked"] = "MEM_IOS_LOGIN_SCREEN_SIGN_IN_WITH_GOOGLE_CLICKED";
    GqTrackedEventSlug["MemIosLoginScreenLearnMoreClicked"] = "MEM_IOS_LOGIN_SCREEN_LEARN_MORE_CLICKED";
    GqTrackedEventSlug["MemIosLoginScreenAccountNotFoundMustSignUpClicked"] = "MEM_IOS_LOGIN_SCREEN_ACCOUNT_NOT_FOUND_MUST_SIGN_UP_CLICKED";
    /** Home */
    GqTrackedEventSlug["MemIosHomeScreenViewed"] = "MEM_IOS_HOME_SCREEN_VIEWED";
    GqTrackedEventSlug["MemIosHomeScreenLoadedResults"] = "MEM_IOS_HOME_SCREEN_LOADED_RESULTS";
    GqTrackedEventSlug["MemIosHomeScreenSawNoResults"] = "MEM_IOS_HOME_SCREEN_SAW_NO_RESULTS";
    GqTrackedEventSlug["MemIosHomeScreenLoadedMoreResults"] = "MEM_IOS_HOME_SCREEN_LOADED_MORE_RESULTS";
    GqTrackedEventSlug["MemIosHomeScreenRefreshedResults"] = "MEM_IOS_HOME_SCREEN_REFRESHED_RESULTS";
    GqTrackedEventSlug["MemIosHomeScreenSuggestionClicked"] = "MEM_IOS_HOME_SCREEN_SUGGESTION_CLICKED";
    /** Search */
    GqTrackedEventSlug["MemIosSearchScreenViewed"] = "MEM_IOS_SEARCH_SCREEN_VIEWED";
    GqTrackedEventSlug["MemIosSearchScreenSearchExecuted"] = "MEM_IOS_SEARCH_SCREEN_SEARCH_EXECUTED";
    GqTrackedEventSlug["MemIosSearchScreenSawNoResults"] = "MEM_IOS_SEARCH_SCREEN_SAW_NO_RESULTS";
    GqTrackedEventSlug["MemIosSearchScreenLoadedMoreResults"] = "MEM_IOS_SEARCH_SCREEN_LOADED_MORE_RESULTS";
    GqTrackedEventSlug["MemIosSearchScreenSuggestionClicked"] = "MEM_IOS_SEARCH_SCREEN_SUGGESTION_CLICKED";
    GqTrackedEventSlug["MemIosSearchScreenOpenedFromNavigationBar"] = "MEM_IOS_SEARCH_SCREEN_OPENED_FROM_NAVIGATION_BAR";
    GqTrackedEventSlug["MemIosSearchScreenOpenedFromHomeBar"] = "MEM_IOS_SEARCH_SCREEN_OPENED_FROM_HOME_BAR";
    /** Share Sheet */
    GqTrackedEventSlug["MemIosShareSheetViewed"] = "MEM_IOS_SHARE_SHEET_VIEWED";
    /** Slice Results (Home + Search Views) */
    GqTrackedEventSlug["MemIosSliceResultsItemClicked"] = "MEM_IOS_SLICE_RESULTS_ITEM_CLICKED";
    /** Details */
    GqTrackedEventSlug["MemIosDetailScreenViewed"] = "MEM_IOS_DETAIL_SCREEN_VIEWED";
    GqTrackedEventSlug["MemIosDetailScreenMemEdited"] = "MEM_IOS_DETAIL_SCREEN_MEM_EDITED";
    GqTrackedEventSlug["MemIosDetailScreenMemComposed"] = "MEM_IOS_DETAIL_SCREEN_MEM_COMPOSED";
    GqTrackedEventSlug["MemIosDetailScreenTopicClicked"] = "MEM_IOS_DETAIL_SCREEN_TOPIC_CLICKED";
    GqTrackedEventSlug["MemIosDetailScreenWebLinkClicked"] = "MEM_IOS_DETAIL_SCREEN_WEB_LINK_CLICKED";
    GqTrackedEventSlug["MemIosDetailScreenLinkedFirestoreNoteClicked"] = "MEM_IOS_DETAIL_SCREEN_LINKED_FIRESTORE_NOTE_CLICKED";
    GqTrackedEventSlug["MemIosDetailScreenMentionedFirestoreUserClicked"] = "MEM_IOS_DETAIL_SCREEN_MENTIONED_FIRESTORE_USER_CLICKED";
    GqTrackedEventSlug["MemIosDetailScreenMentionedFirestoreGroupClicked"] = "MEM_IOS_DETAIL_SCREEN_MENTIONED_FIRESTORE_GROUP_CLICKED";
    GqTrackedEventSlug["MemIosDetailScreenRelatedMemsViewed"] = "MEM_IOS_DETAIL_SCREEN_RELATED_MEMS_VIEWED";
    /** Editing */
    GqTrackedEventSlug["MemIosEditorCarouselCameraClicked"] = "MEM_IOS_EDITOR_CAROUSEL_CAMERA_CLICKED";
    GqTrackedEventSlug["MemIosEditorCarouselPhotoClicked"] = "MEM_IOS_EDITOR_CAROUSEL_PHOTO_CLICKED";
    GqTrackedEventSlug["MemIosEditorCarouselGalleryClicked"] = "MEM_IOS_EDITOR_CAROUSEL_GALLERY_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarGalleryClicked"] = "MEM_IOS_EDITOR_TOOLBAR_GALLERY_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarBulletedListClicked"] = "MEM_IOS_EDITOR_TOOLBAR_BULLETED_LIST_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarChecklistClicked"] = "MEM_IOS_EDITOR_TOOLBAR_CHECKLIST_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarAddTagClicked"] = "MEM_IOS_EDITOR_TOOLBAR_ADD_TAG_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarFormatClicked"] = "MEM_IOS_EDITOR_TOOLBAR_FORMAT_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarH1Clicked"] = "MEM_IOS_EDITOR_TOOLBAR_H1_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarH2Clicked"] = "MEM_IOS_EDITOR_TOOLBAR_H2_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarBoldClicked"] = "MEM_IOS_EDITOR_TOOLBAR_BOLD_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarItalicClicked"] = "MEM_IOS_EDITOR_TOOLBAR_ITALIC_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarUnderlineClicked"] = "MEM_IOS_EDITOR_TOOLBAR_UNDERLINE_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarStrikethroughClicked"] = "MEM_IOS_EDITOR_TOOLBAR_STRIKETHROUGH_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarCodeBlockClicked"] = "MEM_IOS_EDITOR_TOOLBAR_CODE_BLOCK_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarQuoteBlockClicked"] = "MEM_IOS_EDITOR_TOOLBAR_QUOTE_BLOCK_CLICKED";
    GqTrackedEventSlug["MemIosEditorToolbarRemoveFormattingClicked"] = "MEM_IOS_EDITOR_TOOLBAR_REMOVE_FORMATTING_CLICKED";
    /** Side Navigation */
    GqTrackedEventSlug["MemIosSideNavigationViewed"] = "MEM_IOS_SIDE_NAVIGATION_VIEWED";
    /** Help And Support */
    GqTrackedEventSlug["MemIosSupportScreenViewed"] = "MEM_IOS_SUPPORT_SCREEN_VIEWED";
    GqTrackedEventSlug["MemIosSupportScreenProductUpdatesClicked"] = "MEM_IOS_SUPPORT_SCREEN_PRODUCT_UPDATES_CLICKED";
    GqTrackedEventSlug["MemIosSupportScreenFaqsClicked"] = "MEM_IOS_SUPPORT_SCREEN_FAQS_CLICKED";
    GqTrackedEventSlug["MemIosSupportScreenTutorialsClicked"] = "MEM_IOS_SUPPORT_SCREEN_TUTORIALS_CLICKED";
    GqTrackedEventSlug["MemIosSupportScreenBlogClicked"] = "MEM_IOS_SUPPORT_SCREEN_BLOG_CLICKED";
    GqTrackedEventSlug["MemIosSupportScreenContactUsClicked"] = "MEM_IOS_SUPPORT_SCREEN_CONTACT_US_CLICKED";
    /** Other Events */
    GqTrackedEventSlug["MemIosAccountLoggedOut"] = "MEM_IOS_ACCOUNT_LOGGED_OUT";
    /** Recommended */
    GqTrackedEventSlug["EntitiesRecommended"] = "ENTITIES_RECOMMENDED";
})(GqTrackedEventSlug = exports.GqTrackedEventSlug || (exports.GqTrackedEventSlug = {}));
exports.BatchCreateMemsDocument = {
    kind: "Document",
    definitions: [
        {
            kind: "OperationDefinition",
            operation: "mutation",
            name: { kind: "Name", value: "BatchCreateMems" },
            variableDefinitions: [
                {
                    kind: "VariableDefinition",
                    variable: { kind: "Variable", name: { kind: "Name", value: "inputs" } },
                    type: {
                        kind: "NonNullType",
                        type: {
                            kind: "ListType",
                            type: {
                                kind: "NonNullType",
                                type: { kind: "NamedType", name: { kind: "Name", value: "CreateMemInput" } },
                            },
                        },
                    },
                },
            ],
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "batchCreateMems" },
                        arguments: [
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "inputs" },
                                value: { kind: "Variable", name: { kind: "Name", value: "inputs" } },
                            },
                        ],
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "success" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "mem" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [{ kind: "Field", name: { kind: "Name", value: "id" } }],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    ],
};
exports.CreateMemDocument = {
    kind: "Document",
    definitions: [
        {
            kind: "OperationDefinition",
            operation: "mutation",
            name: { kind: "Name", value: "CreateMem" },
            variableDefinitions: [
                {
                    kind: "VariableDefinition",
                    variable: { kind: "Variable", name: { kind: "Name", value: "input" } },
                    type: {
                        kind: "NonNullType",
                        type: { kind: "NamedType", name: { kind: "Name", value: "CreateMemInput" } },
                    },
                },
            ],
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "createMem" },
                        arguments: [
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "input" },
                                value: { kind: "Variable", name: { kind: "Name", value: "input" } },
                            },
                        ],
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "success" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "mem" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [{ kind: "Field", name: { kind: "Name", value: "id" } }],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    ],
};
exports.HealthCheckDocument = {
    kind: "Document",
    definitions: [
        {
            kind: "OperationDefinition",
            operation: "query",
            name: { kind: "Name", value: "HealthCheck" },
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "healthCheckDetails" },
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [{ kind: "Field", name: { kind: "Name", value: "apiStatus" } }],
                        },
                    },
                ],
            },
        },
    ],
};
